Assets/MixedRealityToolkit/InputSystem/Sources/GestureInputSource.cs

```diff
@@ -0,0 +1,191 @@
+
+        /// <summary>
+        /// Manipulation action to use when events are raised.
+        /// </summary>
+        public static MixedRealityInputAction ManipulationAction { get; set; } = MixedRealityInputAction.None;
+
+        /// <summary>
+        /// Navigation action to use when events are raised.
+        /// </summary>
+        public static MixedRealityInputAction NavigationAction { get; set; } = MixedRealityInputAction.None;
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="useRailsNavigation">Should the gesture input source use rails navigation?</param>
+        public GestureInputSource(bool useRailsNavigation) : base("Gesture Input Source")
+        {
+            gestureRecognizer = new GestureRecognizer();
+
+            gestureRecognizer.Tapped += GestureRecognizer_Tapped;
+
+            gestureRecognizer.HoldStarted += GestureRecognizer_HoldStarted;
+            gestureRecognizer.HoldCompleted += GestureRecognizer_HoldCompleted;
+            gestureRecognizer.HoldCanceled += GestureRecognizer_HoldCanceled;
+
+            gestureRecognizer.ManipulationStarted += GestureRecognizer_ManipulationStarted;
+            gestureRecognizer.ManipulationUpdated += GestureRecognizer_ManipulationUpdated;
+            gestureRecognizer.ManipulationCompleted += GestureRecognizer_ManipulationCompleted;
+            gestureRecognizer.ManipulationCanceled += GestureRecognizer_ManipulationCanceled;
+
+            gestureRecognizer.SetRecognizableGestures(GestureSettings.Tap | GestureSettings.ManipulationTranslate | GestureSettings.Hold);
+
+            // We need a separate gesture recognizer for navigation, since it isn't compatible with manipulation
+            navigationGestureRecognizer = new GestureRecognizer();
+
+            navigationGestureRecognizer.NavigationStarted += NavigationGestureRecognizer_NavigationStarted;
+            navigationGestureRecognizer.NavigationUpdated += NavigationGestureRecognizer_NavigationUpdated;
+            navigationGestureRecognizer.NavigationCompleted += NavigationGestureRecognizer_NavigationCompleted;
+            navigationGestureRecognizer.NavigationCanceled += NavigationGestureRecognizer_NavigationCanceled;
+
+            if (useRailsNavigation)
+            {
+                navigationGestureRecognizer.SetRecognizableGestures(GestureSettings.NavigationRailsX | GestureSettings.NavigationRailsY | GestureSettings.NavigationRailsZ);
+            }
+            else
+            {
+                navigationGestureRecognizer.SetRecognizableGestures(GestureSettings.NavigationX | GestureSettings.NavigationY | GestureSettings.NavigationZ);
+            }
+
+            if (gestureRecognizer != null && !gestureRecognizer.IsCapturingGestures())
+            {
+                gestureRecognizer.StartCapturingGestures();
+            }
+
+            if (navigationGestureRecognizer != null && !navigationGestureRecognizer.IsCapturingGestures())
```

Comment:
this can cause BOTH recognizers to be running at the same time as it is possible tyo have the recognizers created and not started at the same time.

Reply:
I'm not sure I understand your comment

Reply:
it is possible for gestureRecognizer to be != null and not recognizing at the same time as the navRecognizer is in the same exact state.

by the code here, each would be started and be running at the same time... which would result in both manipulation and navigation gestures being recognized from the same single gesture

Reply:
in other words, we should never allow gesture recognizers with mutually exclusive settings (manipulation and navigation) to be recognizing at the same time

Reply:
Ah, I see. Then why does the HTK do it?

Reply:
>  Then why does the HTK do it?

I honestly don't know,